### PR TITLE
Typing for pg-cursor

### PR DESCRIPTION
**Pull request by elias-fauser**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by elias-fauser** [comment]
@Tiogshi Would you have time to quickly review this PR?

File: types/pg-cursor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        types?: CustomTypesConfig;
+    }
+    type ResultCallback<RowType> = (err: Error | undefined, rows: RowType[], result: QueryResult) => void;
+}
+
+declare class Cursor<Row = any> {
+    constructor(query: string, values?: any[], config?: Cursor.CursorQueryConfig);
+    submit: (connection: Connection) => void;
+    read: (maxRows: number, callback: Cursor.ResultCallback<Row>) => void;
+    close: (callback: (err: Error) => void) => void;
```
Comment (martin-badin) [inline types/pg-cursor/index.d.ts:28]:
```suggestion
    submit(connection: Connection): void;
    read(maxRows: number, callback: Cursor.ResultCallback<Row>): void;
    read(maxRows: number): Promise<Error | RowType[]>;
    close(callback: (err: Error) => void): void;
    close(): Promise<Error>;
```

File: types/pg-cursor/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        rowMode?: "array";
+        /**
+         * Custom type parsers for just this query result.
+         */
+        types?: CustomTypesConfig;
+    }
+    type ResultCallback<RowType> = (err: Error | undefined, rows: RowType[], result: QueryResult) => void;
+}
+
+declare class Cursor<Row = any> {
```
Comment (martin-badin) [inline types/pg-cursor/index.d.ts:24]: The class should be extended from EventEmitter. Please see the source code
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L9

Comment (martin-badin) [reply types/pg-cursor/index.d.ts:24]: You should define the deprecated function as well.
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L254

File: types/pg-cursor/index.d.ts (hunk 1 above)
Comment (martin-badin) [reply types/pg-cursor/index.d.ts:28]: Methods returns promise if callback is not defined
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L208
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L231

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/pg-cursor/index.d.ts (hunk 1 above)
Comment (elias-fauser) [reply types/pg-cursor/index.d.ts:28]: Thanks for noticing. This was indeed wrong. Added function overloads for the different cases

File: types/pg-cursor/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// Type definitions for pg-cursor 2.5
+// Project: https://github.com/brianc/node-postgres#readme
+// Definitions by: Tiogshi Laj <https://github.com/Tiogshi>, Edited by Elias Fauser <https://github.com/elias-fauser>
```
Comment (amcasey) [inline types/pg-cursor/index.d.ts:3]: This section follows a schema - please remove "Edited by"

**Review by amcasey — CHANGES_REQUESTED** [review]
Any remaining concerns @martin-badin

**Review by martin-badin — APPROVED** [review]

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by elias-fauser** [comment]
Ready to merge

**Comment by tomjrees** [comment]
@elias-fauser Thanks for the types! There is a line which is not working for me, and it looks a little funky.

I think that:

```typescript
import EventEmitter = require("events");
```

should be:

```typescript
import { EventEmitter } from 'events'
```

But I have a weak grasp of the nuances of typescript here. I am using a templated `tsconfig.json` file with     `"esModuleInterop": true` which might be causing my problem? I can't switch it off without breaking the rest of my project though. Thought I'd try posting here to check if there's a special reason for this.

**Comment by elias-fauser** [comment]
@tomjrees I will have a look and see if there will be any issue with changing the import statement

**Comment by elias-fauser** [comment]
@tomjrees I opened the following PR for your issue: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61179

**Pull request merged**